src/coreclr/jit/emitxarch.cpp

```diff
@@ -9128,15 +9121,15 @@ void emitter::emitDispIns(
                 }
                 if ((val > -1000) && (val < 1000))
                 {
-                    printf("%d", val);
+                    printf("%d", (int)val);
                 }
                 else if ((val > 0) || (val < -0xFFFFFF))
                 {
-                    printf("0x%IX", val);
+                    printf("0x%IX", (INT64)val);
                 }
                 else
                 { // (val < 0)
-                    printf("-0x%IX", -val);
+                    printf("-0x%IX", (INT64)-val);
                 }
                 emitDispCommentForHandle(srcVal, id->idDebugOnlyInfo()->idMemCookie, id->idDebugOnlyInfo()->idFlags);
```

Comment:
Since we have the debug info can the `#ifdef DEBUG` in `emitDispCommentForHandle` be removed? Then your example should show the name of the fields.

Reply:
Sure, it's a left-over from my initial impl where I wanted to if-def all `idDebugOnlyInfo()` but quickly realized that I won't be able to even print method names :)